jpype/_core.py

```diff
@@ -104,23 +113,48 @@ def isJVMStarted():
     return _jpype.isStarted()
 
 
-def _hasClassPath(args):
+def _hasClassPath(args: typing.Tuple[_PathOrStr, ...]) -> bool:
     for i in args:
-        if i.startswith('-Djava.class.path'):
+        if isinstance(i, str) and i.startswith('-Djava.class.path'):
             return True
     return False
 
 
-def _handleClassPath(clsList):
+def _handleClassPath(
+    classpath: typing.Union[
+        _PathOrStr,
+        typing.Tuple[_PathOrStr, ...]
+    ],
+) -> str:
+    """
+    Return a classpath which represents the given tuple of classpath specifications
+    """
     out = []
-    for s in clsList:
-        if not isinstance(s, str):
-            raise TypeError("Classpath elements must be strings")
-        if s.endswith('*'):
+
+    if isinstance(classpath, (str, os.PathLike)):
+        classpath = (classpath,)
+    try:
+        # Convert anything iterable into a tuple.
+        classpath = tuple(classpath)
+    except TypeError:
+        raise TypeError("Unknown class path element")
+
+    for element in classpath:
+        try:
+            pth = os.fspath(element)
+        except TypeError as err:
+            raise TypeError("Classpath elements must be strings or Path-like") from err
+
+        if isinstance(pth, bytes):
```

Comment:
the docs of os.PathLike states that bytes are allowed, but not preferred. At least the error message is confusing then

Reply:
The elements must be strings of Path-like. Means it can be ``bytes`` if you know it. I chose not to put the fact that it could be bytes in the error explicitly, for the reason that it isn't preferred and to keep the message succinct.

I can change it to ``strings, bytes, or Path-like`` if you prefer though.